src/javatests/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventRecordConstructorTest.java

```diff
@@ -65,117 +57,59 @@
-    QueryPlan queryPlan = createQueryPlan();
-    HookContext context = createContext(queryPlan);
-    context.setHookType(HookType.PRE_EXEC_HOOK);
+  public void preExecHook_success() {
+    hookContext.setHookType(HookType.PRE_EXEC_HOOK);
+    queryState.setCommandType(HiveOperation.QUERY);
 
     // Act
-    Optional<GenericRecord> record = eventRecordConstructor.constructEvent(context);
+    Optional<GenericRecord> record = eventRecordConstructor.constructEvent(hookContext);
 
     // Assert
-    assertThat(record)
-        .hasValue(
-            new GenericRecordBuilder(QUERY_EVENT_SCHEMA)
-                .set("QueryId", DEFAULT_QUERY_ID)
-                .set("QueryText", DEFAULT_QUERY_TEXT)
-                .set("EventType", "QUERY_SUBMITTED")
-                .set("ExecutionMode", "NONE")
-                .set("StartTime", 1234L)
-                .set("RequestUser", "test_user")
-                .set("UserName", System.getProperty("user.name"))
-                .set("SessionId", "test_session_id")
-                .set("IsTez", false)
-                .set("IsMapReduce", false)
-                .set("InvokerInfo", "test_session_id")
-                .set("ThreadName", "test_thread_id")
-                .set("HookVersion", "1.0")
-                .set("ClientIpAddress", "192.168.10.10")
-                .set("HiveAddress", "hive_addr")
-                .set("HiveInstanceType", "HS2")
-                .set("OperationId", "test_op_id")
-                .set("MapReduceCountersObject", "[]")
-                .set("TezCountersObject", "[]")
-                .build());
+    assertThat(record).hasValue(TestUtils.createPreExecRecord());
   }
 
   @Test
-  public void postExecHook_success() throws Exception {
-    QueryPlan queryPlan = createQueryPlan();
-    HookContext context = createContext(queryPlan);
-    context.setHookType(HookType.POST_EXEC_HOOK);
+  public void postExecHook_success() {
+    hookContext.setHookType(HookType.POST_EXEC_HOOK);
 
     // Act
-    Optional<GenericRecord> record = eventRecordConstructor.constructEvent(context);
+    Optional<GenericRecord> record = eventRecordConstructor.constructEvent(hookContext);
 
     // Assert
-    assertThat(record)
-        .hasValue(
-            new GenericRecordBuilder(QUERY_EVENT_SCHEMA)
-                .set("QueryId", DEFAULT_QUERY_ID)
-                .set("EventType", "QUERY_COMPLETED")
-                .set("EndTime", QUERY_END_TIME)
-                .set("RequestUser", "test_user")
-                .set("UserName", System.getProperty("user.name"))
-                .set("OperationId", "test_op_id")
-                .set("Status", "SUCCESS")
-                .set("PerfObject", "{}")
-                .set("HookVersion", "1.0")
-                .set("MapReduceCountersObject", "[]")
-                .set("TezCountersObject", "[]")
-                .build());
+    assertThat(record).hasValue(TestUtils.createPostExecRecord("SUCCESS"));
```

Comment:
Consider accepting EventStatus enum in createPostExecRecord and casting it to String with `.name()` in the method body.